tests/migrations/test_update_legacy_filtered_positives.py

```diff
@@ -0,0 +1,193 @@
+    ) as mock_split_mongo_samples_by_version:
+        yield mock_split_mongo_samples_by_version
+
+
+@pytest.fixture
+def mock_update_filtered_positive_fields():
+    with patch(
+        "migrations.update_legacy_filtered_positives.update_filtered_positive_fields"
+    ) as mock_update_filtered_positive_fields:
+        yield mock_update_filtered_positive_fields
+
+
+# ----- test migration -----
+
+
+def test_update_legacy_filtered_positives_exception_raised_if_user_cancels_migration(
+    mock_user_input, mock_helper_database_updates, filtered_positive_testing_samples
+):
+    with pytest.raises(Exception):
+        mock_user_input.return_value = "no"
+        mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+
+        update_legacy_filtered_positives.run("crawler.config.integration")
+
+        mock_update_mongo.assert_not_called()
+        mock_update_mlwh.assert_not_called()
+
+
+def test_update_legacy_filtered_positives_exception_raised_if_user_enters_invalid_input(
+    mock_user_input, mock_helper_database_updates, filtered_positive_testing_samples
+):
+    with pytest.raises(Exception):
+        mock_user_input.return_value = "invalid_input"
+        mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+
+        update_legacy_filtered_positives.run("crawler.config.integration")
+
+        mock_update_mongo.assert_not_called()
+        mock_update_mlwh.assert_not_called()
+
+
+def test_update_legacy_filtered_positives_catches_error_connecting_to_mongo(
+    mock_helper_database_updates, mock_v0_version_set
+):
+    with pytest.raises(Exception):
+        with patch(
+            "migrations.update_legacy_filtered_positives.legacy_mongo_samples",
+            side_effect=Exception("Boom!"),
+        ):
+            mock_v0_version_set.return_value = False
+            mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+            update_legacy_filtered_positives.run("crawler.config.integration")
+
+            mock_update_mongo.assert_not_called()
+            mock_update_mlwh.assert_not_called()
+
+
+def test_error_connecting_to_mysql_databases_raises_exception(
+    mock_v0_version_set,
+    mock_helper_database_updates,
+    mock_query_helper_functions,
+    mock_extract_required_cp_info,
+):
+    mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+    mock_legacy_mongo_samples, mock_get_cherrypicked_samples_by_date = mock_query_helper_functions
+
+    mock_v0_version_set.return_value = False
+    mock_legacy_mongo_samples.return_value = [{"plate_barcode": "1"}]
+    mock_extract_required_cp_info.return_value = [["id_1"], ["plate_barcode_1"]]
+    mock_get_cherrypicked_samples_by_date.side_effect = Exception("Boom!")
+
+    with pytest.raises(Exception):
+        update_legacy_filtered_positives.run("crawler.config.integration")
+        mock_update_mongo.assert_not_called()
+        mock_update_mlwh.assert_not_called()
+
```

Comment:
Might also be worth testing no database updates occur when:
* extracting cherrypicking info throws
* getting cherrypicked samples by date throws
* splitting the samples throws
* updating the filtered positive values throws

Reply:
I've added three tests - already had the `get_cherrypicked_samples_by_date` tested, but renamed to make more clear